File: primes/roots.py
from primePy import primes
import matplotlib.pyplot as plt

limit = 100

last = None
gap = 0
n = 0
gaps = []
roots = []

for p in primes.upto(limit):
    n += 1
    if last:
        gap = p - last
    last = p
    if gap:
        root = p ** (1 / gap)
    else:
        root = 0
    gaps.append(gap)
    roots.append(root)
    print(f"{n}. 1/{p}: {1/p}, gap: {gap}, root: {root}")

File: primes/roots.pl
#!/usr/bin/env perl

# TBD

use strict;
use warnings;

use Data::Dumper::Compact qw(ddc);
use Math::BigFloat ();
use Math::Prime::XS qw(primes);

my $limit = shift || 100;

my $last;
my $gap = 0;
my $n = 0;
for my $p (primes($limit)) {
    $n++;
    if (defined $last) {
        $gap = $p - $last;
    }
    $last = $p;
    my $x = Math::BigFloat->new($p);
    my $y = Math::BigFloat->new($gap);
    my $root = $x->broot($y);
    printf "%d. 1/%d: %.4f, gap: %d, root: %.4f\n", $n, $p, 1 / $p, $gap, $root;
}
